File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int letter;
int word;
int sentence;

int main (void)

{
    //ask user input for text
    string text = get_string("What is the text you would like to analyze?: ");

    //define lenght of text
    int n = strlen(text);

    count words
    if (isalnum(text[0]))
    {
        word = 1;
    }

    for (int i = 0; i < n;  i++)
    {
        //count letters

        if (isalnum(text[i]))
        {
            letter++;
        }

        //count words

        if (i < n - 1 && isspace(text[i]) && isalnum(text[i + 1]))
        {
            word++;
        }

        //count sentences

        if (i > 0 && (text[i] == '.' || text[i] == '?' || text[i] == '!') && isalnum(text[i - 1]))
        {
            sentence++;
        }

    }

//apply algorithm for Coleman-Liau index

    float grade = ((0.0588 * (100 * letter / word)) - (0.296 * (100 * sentence / word)) - 15.8);
    int roundgrade = round (grade);

//debugger

    //print f("Letters: %i\n Words: %i\n Sentences: %i\n", letter, word, sentence);


//print result for index
    if (roundgrade <= 1)
    {
        printf("Before Grade 1\n");
    }
    else if (roundgrade < 16)
    {
        printf("Grade %i\n", roundgrade);
    }
    else
    {
        printf("Grade 16+\n");
    }
    

}

File: readability.py
from cs50 import get_string


#ask user input for text
text = get_string("What is the text you would like to analyze?: ")

letters = 0
words = 0 
sentences = 0
counter = 0

for i in text:
    counter += 1

for i in range(counter):
    #count letters using ascii code
    if (ord(text[i]) >= 65 and ord(text[i]) <= 122):
        letters += 1

    #count words by reading spaces
    elif (ord(text[i]) == 32 and (ord(text[i - 1]) != 33 and ord(text[i - 1]) != 46 and ord(text[i - 1]) != 63)):
        words += 1

    #count sentences by finding dots, exclamation marks and interrogatives
    elif (ord(text[i]) == 33 or ord(text[i]) == 46 or ord(text[i]) == 63):
        sentences += 1
        words += 1

L = letters * 100 / words
S = sentences * 100 / words

#Coleman-Liau index is computed using the formula
index = round(0.0588 * L - 0.296 * S - 15.8)

#output the result to the user
if (index < 1):
    print("Before Grade 1")

elif (index >= 16):
    print("Grade 16+")

else:
    print(f"Grade {index}")
